requirements.txt

```diff
@@ -2,4 +2,3 @@ pip<24.0
 pytest<8.0
 filecheck<0.0.24
 lit<16.0.0
-coverage<8.0.0
```

Comment:
isn't coverage an import we use in `xdsl-opt`? here: https://github.com/xdslproject/xdsl/blob/18415559ca727a90a6692a3c38b77d29c53c0a04/xdsl/xdsl_opt_main.py#L5

Reply:
Alternative question, if this is required, why is it not triggering an error on the CI when removing it?

Reply:
Because it is installed in the CI script here.
I think we should not do that as long as it is imported in xdsl-opt.
Removing this dependency in xdsl-opt would be good though: now that frozenlist is gone, coverage is the only non-pure Python dependency 🙂

Reply:
Good catch!
I moved coverage as I felt like this is not something we want a user to install (and run) but rather a developer. You are right though, the way we have set it up in xdsl-opt means that it needs to be imported.

The reason why it worked in both CIs is that both of them set up the developer environment, so it is installed. I will change that s.t. the only-Python one installs the user set up.

Reply:
From the side of removing the dependency, I would be happy to hear suggestions of how we could generate coverage better. I did not manage to do it last time I tried, but there might very well be a way...

Reply:
Especially since you can run coverage from the command line. That might even generate better coverage, but I stuggled with argument handling last time...